tapgarden/caretaker.go

```diff
@@ -533,7 +533,7 @@ func (b *BatchCaretaker) stateStep(currentState BatchState) (BatchState, error)
 		//
 		// TODO(roasbeef): need to invalidate asset creation if on
 		// restart leases are gone
-		ctx, cancel := b.WithCtxQuit()
+		ctx, cancel := b.WithCtxQuitNoTimeout()
```

Comment:
Perhaps the timeout should still exist but be proportional to the amount of work to complete. Future work for another PR perhaps.